app/jobs/launch_promo2018q1_job.rb

```diff
@@ -0,0 +1,23 @@
+# Generates the promo tokens and sends emails
+class LaunchPromo2018q1Job < ApplicationJob
```

Comment:
rake task would be good i think.
it'd generate the tokens and queue up all the emails